test/ConstructiveSolidGeometry/CSG_primitives.jl

```diff
@@ -110,16 +110,52 @@ no_translations = (rotation = one(SMatrix{3, 3, T, 9}), translation = zero(Carte
             # Internally, the primitive stores the offset of phi in a rotation, but it should not be part of the config file
             @test !haskey(output["difference"][1], "rotation") 
             @test t.a.rotation ≈ Geometry(T, output, default_units, no_translations).a.rotation
-            
+        
             # No warnings or errors when decomposing the Torus into surfaces
             @test_nowarn CSG.surfaces(t.a)
             @test_nowarn CSG.surfaces(t.b)
-
+        
             # Check if all Torus are saved the right way
             @test t.a isa CSG.Torus
             @test t.b isa CSG.Torus
         end
-    end
+    
+        # Test different constructor versions and dictionary construction
+        torus1 = @inferred CSG.Torus(CSG.ClosedPrimitive, r_torus = 1f0, r_tube = 1f0)
+        torus2 = @inferred CSG.Torus{Float32}(r_torus = 1.0, r_tube = 1.0)
+        @test torus1 == torus2
+        dict = Dict("torus"   => Dict(
+                "r_torus"       => 1.0,
+                "r_tube"        => 1.0))
+                # "phi"           => π,
+                # "theta"         => (π,2π)))
+        torus4 = Geometry(T,dict,default_units,no_translations)
+        output = Dictionary(torus4)
+        @test dict == output
+
+        ## Test in method for several tori
+        # Non transformed geometries
+        @test !in(CartesianPoint{Float32}(0,0,0.5),torus1)
+        @test in(CartesianPoint{Float32}(0,0,0),torus1)
+        torus1_open = @inferred CSG.Torus(CSG.OpenPrimitive, r_torus = 1f0, r_tube = 1f0)
+        @test in(CartesianPoint{Float32}(0,2,0),torus1)
+        @test !in(CartesianPoint{Float32}(0,2,0),torus1_open)
+        # Boundary behaviour of closed and open primitives for transformed 
+        # geometries (origin, rotation)
+        torus_closed_trafo = @inferred CSG.Torus(CSG.ClosedPrimitive,r_torus=2.0, r_tube=3.0, φ = π, origin=CartesianPoint{Float32}(0,0,1),rotation=SMatrix{3}(0,0,-1,0,1,0,1,0,0))
+        torus_open_trafo = @inferred CSG.Torus(CSG.OpenPrimitive,r_torus=2.0, r_tube=3.0, φ = π, origin=CartesianPoint{Float32}(0,0,1),rotation=SMatrix{3}(0,0,-1,0,1,0,1,0,0))
+
+        @test in(CartesianPoint{Float64}(1,0,-1),torus_closed_trafo)
+        @test !in(CartesianPoint{Float64}(1,0,-1),torus_open_trafo)
+        @test in(CartesianPoint{Float64}(0,0,0),torus_closed_trafo)
+        @test !in(CartesianPoint{Float64}(0,0,0),torus_open_trafo)
+        tol = 1e-8
+        @test !in(CartesianPoint{Float64}(0,-tol,0),torus_closed_trafo)
+        @test in(CartesianPoint{Float64}(0,+tol,0),torus_open_trafo)
+        #Test all valid keyword argument types
+        @inferred CSG.Torus{Float64}(φ=nothing,θ=nothing)
+        CSG.Torus{Float64}(φ=π,θ=(π,2π)) #inferred does not work here, due to evaluation of _get_conemantle_type(θ) during runtime
```

Comment:
Can we change the implementation of `_get_conemantle_type` such that we dont have an `if`-`else`-statement at runtime to make `@inferred` pass?

Reply:
Hmm, the issue is that the surface type depends on values of `θ` and not the type of `θ`...

This could be "solved" by removing `TT1` and `TT2` from `Torus`. 
But then we shift this type instability to the function `surfaces(t::Torus)`... 

I think `:inwards`, `:outwards` could be managed without type instabilities. 
About the `:flat` case I am not sure though. I will look a bit more into that.

Reply:
But I think this falls outside the scope of this PR as this is a discussion 
about the general implementation of the structs `Torus`, `ConeMantle` and the functions `surfaces` or `TorusThetaSurface`.

I would say we open an dedicated issue for this and accepts that `@inferred` does not work for this case for now.

Reply:
I addressed this issue in #287.

So in this PR, this issue can be ignored.